patterns/callback/call_me_back/src/main.nr

```diff
@@ -0,0 +1,133 @@
+mod interfaces;
+
+/*
+    Callback pattern
+    =====================
+    This is an example of callback patterns in Aztec.
+
+    In this example, the callback effect is to set a value note for the original sender.
+
+    Two types of callbacks are demonstrated: atomic and async.
+
+    There are 2 contracts and 2 parties involved:
+    - Alice: initiate the flow and receive the effect of the callback
+    - Bob: trigger the logic on the Callbacker contract, if the callback is async (see below)
+    - CallMeBack contract: initiate the flow, calls Callbacker and receive the callback
+    - Callbacker contract: either store a pending callback (later resolved by bob) or trigger the callback immediately (if the callback is atomic)
+
+    The atomic callback flow is as follow: Alice calls new_atomic_call on CallMeBack contract, which calls call_me_now on the resolver contract, which calls the callback function on CallMeBack contract.
```

Comment:
Can we separate this into enumerable steps so it's easier to reason about it? Both flows